.github/workflows/license_finder.yml

```diff
@@ -36,7 +36,9 @@ jobs:
       - name: Generate requirements.txt (exclude dev dependencies)
         run: |
           poetry export -f requirements.txt --without-hashes > requirements.txt
+          pip install -r requirements.txt
 
       - name: Run license finder
         run: |
-          poetry run license_finder --python-version=3
+          /opt/hostedtoolcache/Ruby/*/x64/lib/ruby/gems/*/gems/license_finder-*/bin/license_finder_pip.py requirements.txt
+          license_finder
```

Comment:
This command exits with an exit code of 0, which is why running `license_finder_pip.py` in the previous step is still necessary.

Reply:
Huh, that's interesting. How did you test it? I just ran `license_finder` locally and got an exit code of 1 with unknown dependencies.

Reply:
I did it before [here](https://github.com/viamrobotics/viam-python-sdk/actions/runs/10405240710/job/28815550358).

In case it was me doing it wrong, I tried it again but it [failed without actually failing the workflow again](https://github.com/viamrobotics/viam-python-sdk/actions/runs/10406128924/job/28818455686?pr=706).

Reply:
This looks like a slightly different failure, where it doesn't even recognize what the dependencies _are_. I'm guessing the linked failures are not running on `3.11.5`? 
In cases where `license_finder` can successfully see dependencies and run, I'd expect that we should see an appropriate failure. In local testing in the flutter sdk e.g., I get the following output

```
LicenseFinder::Pub: [32mis active for '/Users/ethanrodkin/Documents/code/viam/viam-flutter-sdk'[0m

Dependencies that need approval:
_fe_analyzer_shared, 67.0.0, unknown
analyzer, 6.4.1, unknown
...
```

and running `echo $?` returns `1`.

Reply:
I'm not sure what you mean by slightly different failure. `ModuleNotFoundError: No module named 'pip._vendor.six'` is the error we've been getting in the Python SDK.

I forgot to mention, I did print out the exit code [here](https://github.com/viamrobotics/viam-python-sdk/actions/runs/10405355402/job/28815903802), and you can see running returns 0 even though it failed.

Reply:
> I'm not sure what you mean by slightly different failure. `ModuleNotFoundError: No module named 'pip._vendor.six'` is the error we've been getting in the Python SDK.

So what I mean is, there are broadly three classes of response here:
1. `license_finder` runs successfully, all licenses are a-okay and we return `0` (good!)
2. `license_finder` runs successfully, one or more licenses are unknown or invalid and we return `1` (for the purposes of this test, also good!)
3. we get the `ModuleNotFound` error that you have encountered and return `0` (bad!)

My claim is that we want to get away from case 3. When I say "slightly different failure" what I mean is, you're pointing to the case 3 failure, but I am describing encountering a "slightly different failure" of case 2 locally.

The reason I ask about being on `3.11.5` is because, when we had python at `3.11.5` and just ran `license_finder` without anything special, we found ourselves in case 1. When moving to a different version of python, we get into case 3. This makes me believe that other versions of python are creating inconsistencies that prevent `license_finder` from running properly; I believe this is supported by the fact that the error we are seeing is a `pip3` error, not a `license_finder` one.

Therefore, I believe that being on `3.11.5` and just running `license_finder`, which puts us in case 1, is the ideal state and what we should be merging. I believe if in that state you were to add an unlicensed dependency in `requirements.txt` and run the test, you would find yourself in case 2, _not_ in case 3.

Reply:
This makes sense to me. Thanks for taking the time to explain that, Ethan!